x-pack/plugins/security_solution/public/entity_analytics/common/get_start_date_from_risk_score.ts

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import dateMath from '@kbn/datemath';
+import moment from 'moment';
+
+/**
+ * return start date of risk scoring by calculating the difference between risk score timestamp and risk range start date
+ * return the same risk range start date if it's a date
+ * return now-30d  if there erros in parsing the date
+ */
+export const getStartDateFromRiskScore = ({
+  riskRangeStart,
+  riskScoreTimestamp,
+}: {
+  riskRangeStart: string;
+  riskScoreTimestamp: string;
+}): string => {
+  try {
+    if (moment(riskRangeStart).isValid()) {
```

Comment:
Under what circumstances would this be true? I assume only if the user has manually adjusted the `RiskEngineConfiguration.range.start` field within the saved object?

Reply:
Yes, the chance is small, bit if client decide to change saved object, and set to specific date